site/en/blog/longer-esw-lifetimes/index.md

```diff
@@ -35,11 +36,12 @@ Unfortunately, this behavior did not apply to extension events. Extension events
 
 ## What's changed
 
-As of Chrome 110, all events reset the lidle timer and the idle timeout will not occur if there are pending events. In other words, assuming there are no unexpected interruptions, extension service workers will now stay alive as long as they are actively processing events.
+As of Chrome 110, all events reset the lidle timer and the idle timeout will not occur if there are pending events. In other words, assuming there are no unexpected interruptions, extension service workers will now typically stay alive as long as they are actively processing events. Be aware that:
 
-While this change removes the five minute limit on extension service worker lifetimes, you are still strongly encouraged to design your extensions to be event-based and guard against unexpected termination. Not only should you refrain from keeping your extensions alive indefinitely (which we consider an anti-pattern), you should test your extensions to ensure that they're not doing this unintentionally.
+* The service worker will terminate after 30 seconds if there are no events.
+* The service worker will terminate after 5 minutes if a single request, such as an event or API call, takes longer than 5 minutes to process (with the exception of a native messaging connection).
 
-In a future release, we may change how specific events are processed to prevent keeping service workers alive indefinitely, but without shutting down long-running work. In the meantime, make sure your extensions yield when possible and prepare for termination by persisting extension states. 
+You are still strongly encouraged to design your extensions to be event-based and guard against unexpected termination. Not only should you refrain from keeping your extensions alive indefinitely (which we consider an anti-pattern), you should test your extensions to ensure that they're not doing this unintentionally. Make sure your extensions yield when possible and prepare for termination by persisting extension states.
```

Comment:
This does not quite fit the rest of the flow: "You are still strongly encouraged to design your extensions to be event-based and guard against unexpected termination."

Reading the post I see nothing that would indicate that sw's are no longer event-based (as we don't explicitly mention hacking the lifecycle).

Reply:
I remember no. Simeon strongly insisted on this line. We can get rid of it. @oliverdunk I'd be curious to know, if you have any opinions on this line.

Reply:
I think the two important things to communicate are:

- We may change this behaviour to be stricter in the future (though hopefully in a way that does not impact genuine extensions).
- Your extension may terminate unexpectedly even with these changes (for example, if the browser crashes).

How about something like:

> In a future release, we may continue to iterate on this behaviour to ensure extension service workers are terminated when possible, without shutting down long-running work. As part of building a resource conscious extension, extension service workers should always yield when possible. Additionally, extensions should prepare for unexpected termination by persisting state. This makes sure they are robust against unpredictable events like the browser being forcefully closed by the user.

I think that's a bit wordy but better in some sense?

Reply:
I just submitted a change with a version of Oliver's suggestion.